src/mpid/ch4/src/ch4_impl.h

```diff
@@ -455,6 +455,7 @@ static inline int MPIDI_CH4I_valid_group_rank(MPIR_Comm * comm, int rank, MPIR_G
     do {                                                        \
         mpi_errno = MPID_Progress_test();                       \
         if (mpi_errno != MPI_SUCCESS) MPIR_ERR_POP(mpi_errno);  \
+        MPID_THREAD_CS_YIELD(GLOBAL, MPIR_THREAD_GLOBAL_ALLFUNC_MUTEX); \
```

Comment:
Is it better to have a comment why we need `global_lock yield` after NM/Device progress?

Reply:
I added a TODO comment. We need to look for the best place and frequency to yield that lock. For now, this is a simple temporary solution.